### PR TITLE
[CSFS-153] Article Publishing Endpoint

### PR DESCRIPTION
**Pull request by modupeadeonojobi**
Screenshot (113)

### CODE REVIEW

File: submissions/modupeAdeOnojobi/slim/publishing_house/app/controllers/ArticleController.php [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+
+use App\Model\Article;
+use App\Model\Admin;
+use Psr\Http\Message\ResponseInterface as Response;
+use Psr\Http\Message\ServerRequestInterface as Request;
+
+class ArticleController
+{
+    #GET ALL ARTICLES
```
Comment (adaugochi) [inline submissions/modupeAdeOnojobi/slim/publishing_house/app/controllers/ArticleController.php:13]: The task didn't ask for this endpoint

File: submissions/modupeAdeOnojobi/slim/publishing_house/app/controllers/ArticleController.php [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    public function index(Request $request, Response $response, $args)
+    {
+        $articles = Article::where('articles.id', '>', 0)->join('admins', 'articles.created_by', 'admins.id')
```
Comment (adaugochi) [inline submissions/modupeAdeOnojobi/slim/publishing_house/app/controllers/ArticleController.php:16]: Your where condition here is not supposed to be. `id` won't start from `0`

File: submissions/modupeAdeOnojobi/slim/publishing_house/app/models/admin.php [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+{
+    // protected $fillable = ['title', 'description', 'status', 'created_by', 'created_at'];
+    // public $timestamps = false;
+
+    // public function admin() {
+
+    //     return $this->belongsTo('admin', 'created_by');
+
+    // }
+
```
Comment (adaugochi) [inline submissions/modupeAdeOnojobi/slim/publishing_house/app/models/admin.php:18]: Remove comments when no in use

**Review by adaugochi — CHANGES_REQUESTED** [review]
Don't push `.env` just push a sample of the .env file. Don't push `composer.lock` file. Add it to .env

**Comment by adaugochi** [comment]
Well done

File: submissions/modupeAdeOnojobi/slim/publishing_house/app/models/admin.php (hunk 3 above)
Comment (modupeadeonojobi) [reply submissions/modupeAdeOnojobi/slim/publishing_house/app/models/admin.php]: Done, kindly review my pull request.

**Review by adaugochi — CHANGES_REQUESTED** [review]
Also, add composer.lock file in your gitignore. I am still seeing your env file as part of your PR. To make this changes reflect, you have to delete it and do an git add and create the file again. Only push you sample env file

**Comment by modupeadeonojobi** [comment]
I have done it.

**Review by adaugochi — CHANGES_REQUESTED** [review]
I am still seeing .env 

image
 What I want to see is a sample of your .env file, not  the .env file itself

**Comment by modupeadeonojobi** [comment]
Can you still see it?

**Comment by modupeadeonojobi** [comment]
I really don't get what you want me to do.
You said that I should remove .env file and I did just that.
How do I add composer.lock file to a file that I has already be removed?

**Comment by adaugochi** [comment]
> I really don't get what you want me to do.
> You said that I should remove .env file and I did just that.
> How do I add composer.lock file to a file that I has already be removed?

Sorry, I actually meant that `composer.lock` file should not be part of your PR. Add the `composer.lock` file to `.gitignore`. Same goes with `vendor` and `.env` file. There are not suppose to be part of your PR. They are not meant to be push to the outside

**Comment by modupeadeonojobi** [comment]
I have done it.

**Review by adaugochi — APPROVED** [review]

**Pull request merged**